_posts/2024-10-03-pixi-basics.md

```diff
@@ -0,0 +1,411 @@
+    1.49 ± 0.03 times faster than 'conda create -y -p condaenv ivar && rm -rf condaenv'
+
+```
+
+</details>
+
+For the complicated installation, pixi is 2.21x faster than conda.
+For individual installations, it is consistently more than 1.4x faster than conda.
+
+## Day to day operations
+
+Why would you even go through all this?
+If you have read this post up to here, I am guessing that you already use conda and do not appreciate being told to fix what isn't broken.
+
+Instead, consider that there could be times when it is way more appropriate to use pixi.
+For example, let's say that you are working in a project folder where you make your own "lab notebook" with markdown.
+The project is simple; it is reads mapped to a reference genome.
+
+What were your methods? Your lab notebook/readme says that you downloaded `SRR123` using `fastq-dump`, downloaded `NC_123`, and then mapped the reads with `bwa`. Finally, you sorted and indexed your bam file with `samtools`.
+
+In most of our notebooks, we might have forgotten to specify what software we used or maybe what versions.
+The most careful of us might have had a conda environment file.
+The 0.01% of us might have even been careful enough to save the environment into the actual project folder.
+
+However, with pixi, the environment creation is fast enough that you don't have to save it forever in your project folder.
+The initialization is folder/project based too -- it is simple enough to write into your notebook to start with `pixi shell` and to have the `pixi.toml` file as part of your project.
+This workflow gets much better if you are in a collaborative environment with multiple hands on the same folder.
+Activating the correct environment is pretty much fail-safe because the folder itself dictates the environment.
+Or, dare to imagine, if/when you archive a bioinformatics project, all the software that you used could be freezed and specified in a single toml file.
+
+That is all to say that I don't think that there is a definitive reason to switch to pixi -- I think it just flows better.
+The same way that you start off a development project with `git init`.
+
+## Caveats
+
+There are many caveats to switching over from conda.
+In fact, there is a [whole article](https://pixi.sh/latest/switching_from/conda/) about it.
+Here are some noteworthy caveats.
+
+* Python itself should be added to the dependencies section, e.g.,
+
+```toml
+[dependencies]
+python = ">=3.9"
+```
+
+* pip requirements can be added under the `[project]` section, e.g.,
+
+```toml
+[project]
+dependencies = [
+    "numpy",
+    "pandas",
+    "matplotlib",
+]
+```
+
+* Specifically for Python, you can specify the required version with `requires-python` like so:
+
+```toml
+[project]
+requires-python = ">=3.9"
+```
+
+* Adding pypi packages on the command line is straightforward too, e.g.,
+
+```bash
+# allows you to add a library to your python script, e.g., 
+#   from rich import print
+pixi add --pypi rich
+```
```

Comment:
Add a few sentence conclusion/sales pitch for pixi

Reply:
I added at the end:

> ## conclusions
> I wanted to write all this to give some overview and show why someone might want to switch to pixi in some instances.
I was expecially motivated due to our own local computational architecture.
However, I do not want to make it sound like this is a total replacement for Conda -- I can already foresee using a mix of both in my own day-to-day.
Pixi feels much more like a collaborative method when working on projects
Or also, a good archival method of keeping the same software and versions in a single project.
Being fast is also not a bad feature too ;)  
That all said, I hope this encourages people to try it out!

Reply:
Thank you for that suggestion